canonical/cloud-init
# Add sshd-keygen disable drop-in conf

## Proposed Commit Message


```
Inhibit sshd-keygen@.service if cloud-init is active

In some cloud-init enabled images the sshd-keygen@.service
may race with cloud-init and prevent ssh host keys from being
generated or generating host keys twice slowing boot and  consuming
additional entropy during boot.  This drop-in unit adds a condition to
the sshd-keygen@.service which prevents running if cloud-init is active.

```

## Additional Context


## Test Steps

1. boot centos/rhel based cloud-init image
2.  check systemctl status sshd-keygen@ecdsa to see if service ran (it should run)
3.  install/upgrade cloud-init package from this PR
4.  cloud-init clean --logs ; rm -f /etc/ssh/ssh_host*; reboot
5.  check systemctl status sshd-keygen@ecdsa to see if service ran (it should NOT run)

## Checklist:
